addons/l10n_ch/models/ir_actions_report.py

```diff
@@ -27,11 +27,11 @@ def apply_qr_code_ch_cross_mask(self, width, height, barcode_drawing):
         qr_cross = ReportLabImage((width/2 - cross_width/2) / mm, (height/2 - cross_height/2) / mm, cross_width / mm, cross_height / mm, cross_path.as_posix())
         barcode_drawing.add(qr_cross)
 
-    def _render_qweb_pdf_prepare_streams(self, data, res_ids=None):
+    def _render_qweb_pdf_prepare_streams(self, report_ref, data, res_ids=None):
         # OVERRIDE
-        res = super()._render_qweb_pdf_prepare_streams(data, res_ids)
-        if res_ids and self.report_name in ('account.report_invoice_with_payments', 'account.report_invoice'):
-            invoices = self.env[self.model].browse(res_ids)
+        res = super()._render_qweb_pdf_prepare_streams(report_ref, data, res_ids)
+        if res_ids and (report := self._get_report(report_ref)).report_name in ('account.report_invoice_with_payments', 'account.report_invoice'):
```

Comment:
We need to change the minimum required python version then.
https://github.com/odoo/odoo/blob/aaa7336a1fcee6ec0157fa3ed4cd3010b3416069/odoo/__init__.py#L19

And we also can simplify the [requirements.txt file](https://github.com/odoo/odoo/blob/master/requirements.txt)

Reply:
@d-fence what is the min python version of 16.0?
The walrus operator was used to avoid changing the indentation but not sure if it's worth dropping support for a version of python